how-to-guides/celestia-app-security.md

```diff
@@ -0,0 +1,251 @@
+    server custom 9099 udp/9099 default accept
+
+    # Allow traffic on port 2121 for both TCP and UDP (separate rules for TCP and UDP)
+    server custom 2121 tcp/2121 default accept
+    protection syn-floods 20/sec 40  # Protect against SYN flood attacks
+    server custom 2121 udp/2121 default accept
+    protection syn-floods 20/sec 40  # Protect against SYN flood attacks
+
+    # Allow all traffic from localhost without specifying ports
+    server all accept src 127.0.0.1
+
+    # Allow all traffic from internal network without specifying ports
+    server all accept src 192.168.1.0/24
+
+    # Allow outgoing traffic for general internet access
+    client all accept
+
+# Block all IPv6 traffic
+interface4 enp5s0 ipv6
+    server all drop
+```
+
+::: tip
+Currently bind to eth0, make sure you bind it to the correct interface or bind all
+```interface any world```
+:::
+
+### Key Configuration Details
+
+- P2P (26656): Allows inbound P2P traffic, essential for Celestia consensus node synchronization.
+- RPC (26657): Open to external sources with rate limiting to prevent abuse.
+- Prometheus (26660): Restricted to internal networks or trusted IPs for security.
+- gRPC (9090, 9091): Opened for gRPC communication.
+- REST API (1317): Opened for API access.
+- Blocked IPs: Dynamic IP blocking is applied using blacklists like Firehole Level4 and Emerging Threats.
+
+## Set Up Cron to Automatically Update Blocked IPs
+
+To maintain an updated list of blocked IPs, set up a cron job to fetch the latest blacklists every hour. Here’s how you can configure it.
+
+### Create a Script to Update the Blocked IPs
+
+Create a script in ```/usr/local/bin/update-blocked-ips.sh``` to download and update the IP blacklist.
+
+```sh
+#!/bin/bash
+# Define log file
+LOG_FILE="/var/log/blocklist_update.log"
+
+# Logging function to append messages to log file
+log() {
+    echo "$(date '+%Y-%m-%d %H:%M:%S') - $1" >> "$LOG_FILE"
+}
+
+# Temporary files for downloading the blocklists
+tmp_emerging=$(mktemp) || exit 1
+tmp_firehol=$(mktemp) || exit 1
+tmp_nets=$(mktemp) || exit 1
+
+# Download the block lists from Emerging Threats and FireHOL Level 4
+wget -O $tmp_emerging "http://rules.emergingthreats.net/fwrules/emerging-Block-IPs.txt"
+if [ $? -ne 0 -o ! -s $tmp_emerging ]; then
+    log "Failed to download Emerging Threats blocklist"
+    rm $tmp_emerging
+    exit 1
+fi
+
+wget -O $tmp_firehol "https://iplists.firehol.org/files/firehol_level4.netset"
+if [ $? -ne 0 -o ! -s $tmp_firehol ]; then
+    log "Failed to download FireHOL Level 4 blocklist"
+    rm $tmp_firehol
+    exit 1
+fi
+
+# Preprocess the blocklists: Extract only network entries (lines with '/')
+grep '/' $tmp_emerging > $tmp_nets
+cat $tmp_firehol >> $tmp_nets
+
+# Update the IPSet collections using FireHOL
+firehol ipset_update_from_file blocked_nets nets $tmp_nets
+
+# Clean up temporary files
+rm $tmp_emerging $tmp_firehol $tmp_nets
+log "Successfully updated blocked IPs and networks"
+```
```

Comment:
The current logging level is sufficient as this is an initial script, we will have to find a better approach to download these and sync rather than an SH command. Users can further check iptables once these blacklisted ips are populated